cargo-near-build/src/near/build/mod.rs

```diff
@@ -162,19 +162,37 @@ pub fn run(args: Opts) -> eyre::Result<CompilationArtifact> {
         color,
     )?;
 
-    wasm_artifact.path = crate::fs::copy(&wasm_artifact.path, &out_dir)?;
```

Comment:
sometimes the final destination file is written with non-`wasm-opt`-ed content, 
which might be the reason of failures in concurrent context when many tests run
`near_workspaces::compile_project`

Reply:
after introducing and intermediate stage `[project_name]_optimized.wasm`,  all tests will 
likely read from the same file (from `near_workspaces::compile_project`), which is equal to `[project_name]_optimized.wasm` , 

and not interchangeably from `[project_name].wasm` or `[project_name]_optimized.wasm`

Reply:
was able to easily reproduce this on this commit https://github.com/dj8yfo/near-sdk-rs/commit/027ce75a67a023432901c5eb2d70aa68a276c423

```bash
    receipts: [
        ExecutionOutcome {
            transaction_hash: 61H3aajosM8kDV3jbpuzFaAnMrTCb4DL871gVbbocNag,
            block_hash: HVzmo5HTPZJySPhG5CXMPvde5kMTAGigWKgREbttsoBH,
            logs: [],
            receipt_ids: [
                Dm63UmQr9mqrF6mDiRUxHTaWo8ESKbPHr1p9taEzSzaf,
            ],
            gas_burnt: NearGas {
                inner: 888148937360,
            },
            tokens_burnt: NearToken {
                inner: 88814893736000000000,
            },
            executor_id: AccountId(
                "dev-20241216183040-18506921751143",
            ),
            status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(CompilationError(PrepareError(Deserialization))) })),
        },
    ],
    status: Failure(ActionError(ActionError { index: Some(0), kind: FunctionCallError(CompilationError(PrepareError(Deserialization))) })),
}
thread 'test_storage_deposit_refunds_excessive_deposit' panicked at tests/workspaces.rs:86:5:
assertion failed: res.is_success()


failures:
    simulate_transfer_call_when_called_contract_not_registered_with_ft
    simulate_transfer_call_with_promise_and_refund
    test_close_account_empty_balance
    test_storage_deposit_refunds_excessive_deposit

test result: FAILED. 9 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out; finished in 34.43s
```

Reply:
adding https://github.com/near/near-sdk-rs/pull/1277/commits/ffab607525f02999ea82ed542ea051f2281d0dbd on top of previous change resolved flaky failures

Reply:
corresponding run in near-sdk ci - https://github.com/near/near-sdk-rs/actions/runs/12359514121